src/plots/cartesian/axes.js

```diff
@@ -908,7 +910,9 @@ axes.calcTicks = function calcTicks(ax, opts) {
     var calendar = ax.calendar;
     var ticklabelstep = ax.ticklabelstep;
     var isPeriod = ax.ticklabelmode === 'period';
-
+    var isReversed = ax.range[0] > ax.range[1];
+    var ticklabelIndex = (Array.isArray(ax.ticklabelindex) || !ax.ticklabelindex) ?
```

Comment:
Please use `Lib.isArrayOrTypedArray` instead of `Array.isArray` to support [typed arrays](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Typed_arrays).

Reply:
Also it would better to test `ax.ticklabelindex` first for better performance.
```suggestion
    var ticklabelIndex = (!ax.ticklabelindex || Lib.isArrayOrTypedArray(ax.ticklabelindex)) ?
```